### DISCUSSION TITLE
Requirement to show Full Path to Term in the Search Results Handlebars Template

### DISCUSSION THREAD

**Discussion by agaskelluk** [microsoft-search/pnp-modern-search#1360] [Q&A] [answered]
I am currently using the following handlebars template code to display the tags for the search results:

`<div class="template_tags example-themePrimary">
                                            {{#if owstaxidmetadataalltagsinfo}}
                                                <i class="ms-Icon ms-Icon--Tag" aria-hidden="true"></i>
                                                {{#each (split owstaxidmetadataalltagsinfo ",") as |tag| }}                                                    
                                                    {{trim tag}};
                                                {{/each}}
                                            {{/if}}
                                        </div>`
This issue is that for Hierarchical terms the parent terms are not shown. Users are using Refiners of the form: 
"parentTerm: childTerm" but in the tags displayed in the Search Results only "childTerm" is shown. 

I looked at the debug template and I can see that "owstaxidmetadataalltagsinfo" does indeed only contain the final term in the hierarchy. I want to be able to display the full path to the term in the Search results.

I did notice that owstaxidmetadataalltagsinfoTaxId contains the full path, but it also has a lot of other unwanted detail e.g.
"owstaxidmetadataalltagsinfoTaxId": "L0|#03b66ce74-8fd9-452c-b16d-14b60594628e|PSX:LCR;L0|#0e5f442e5-1a8e-46dc-98f7-a6facbb3e11c|PSX:LNP;L0|" so I was not sure how to split out "PSX:LNP" assuming this is the solution.

What is the best solution please?

Thanks.

**Comment by wobba** [comment]
If you set the column in SharePoint to display the full term path, and map ows_columName to a RefinableString managed property, then you can use this in your template to get: `Path1:Path2:Path3`.

owstaxidmetadataalltagsinfo does not necessarily contain all info needed afaik to show a full path and doing extra API calls to get this information is a bit expensive per result - but you can create extensibility as a web component to do this yourself.

If you believe all is there you can string split any way you want :) Remember that if you have multiple metadata columns on an item, they will all be populated in owstaxidmetadataalltagsinfo.

**Comment by agaskelluk** [comment, marked as answer]
@wobba Thanks for the response. So I assume I need to add my RefinableString managed property to the "Selected Properties" for the web part? 
Would my handlebars template be changed to this:

`<div class="template_tags example-themePrimary"> {{#if **RefinableStringXX**}} <i class="ms-Icon ms-Icon--Tag" aria-hidden="true"></i> {{#each (split **RefinableStringXX**",") as |tag| }} {{trim tag}}; {{/each}} {{/if}} </div>`

Thanks.

**Reply by wobba** [reply]
You need to add the property to Selected Properties yes - and then you manipulate it as needed once you inspect it.

**Reply by agaskelluk** [reply]
@wobba thanks for the update. I did indeed try this approach and it worked great! I appreciate the time you took to respond.
